pkg/operation/botanist/component/dependencywatchdog/bootstrap.go

```diff
@@ -362,17 +197,180 @@ func (b *bootstrapper) WaitCleanup(ctx context.Context) error {
 	return managedresources.WaitUntilDeleted(timeoutCtx, b.client, b.namespace, b.name())
 }
 
+func (b *bootstrapper) deleteOldDWDManagedResources(ctx context.Context) error {
+	log := logf.FromContext(ctx)
+	log.Info("Cleaning up old out of support managed resources for DWD")
+	if err := managedresources.DeleteForSeed(ctx, b.client, b.namespace, ManagedResourceDependencyWatchdogProbe); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("could not delete managed resource %s: %w", ManagedResourceDependencyWatchdogProbe, err)
+	}
+	if err := b.waitUntilManagedResourceDeleted(ctx, ManagedResourceDependencyWatchdogProbe); err != nil {
+		return err
+	}
+
+	if err := managedresources.DeleteForSeed(ctx, b.client, b.namespace, ManagedResourceDependencyWatchdogEndpoint); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("could not delete managed resource %s: %w", ManagedResourceDependencyWatchdogEndpoint, err)
+	}
+	if err := b.waitUntilManagedResourceDeleted(ctx, ManagedResourceDependencyWatchdogEndpoint); err != nil {
+		return err
```

Comment:
Would it make sense to delete both managed resources first and then wait for them to be deleted?

Reply:
That would make waiting for them in parallel , which would need go-routines and channels. So to keep the code simple I did it this way, as its just 2 mr

Reply:
I tried using `managedresources.WaitUntilListDeleted` but it can only be used when we can filter out the desired mr using field or label selectors.
Label selector can't be used because the DWD probe and endpoint mr don't have any common unique labels
Field selector can't be used as it doesn't support `or` , `in` operator , and only supports `=` , `==` and `!=` operator which is not sufficient . Also the field selector is only limited to selecting on fields `spec.name` and `spec.namespace` so no other fields could be used.

So I don't think we could wait for the old mr resources paralelly.

Reply:
You could also add a unique meaningful label and then use it?